src/empty_directory.rs

```diff
@@ -0,0 +1,31 @@
+// http://rosettacode.org/wiki/Empty_directory
+use std::fs;
+use std::path::Path;
+use std::env::args;
+
+fn main() {
+
+    // Open path
+    let path = if let Some(arg) = args().nth(1) {
+        arg
+    } else {
+        println!("You must give a path as argument");
+        return;
+    };
+
+    let dir = Path::new(&path);
+
+    let paths = match fs::read_dir(dir) {
+        Err(e) => {
+            println!("Error: {}", e);
+            return;
+        },
+        Ok(f) => f,
+    };
+
+    if paths.count() > 0 {
+        println!("Directory is not empty");
+    } else {
+        println!("Directory is empty");
+    }
```

Comment:
@aochagavia Thank you for your feedback :)

It is indeed a lot shorter and nicer ! 

I am reading the docs right now (because I never really used the map function before) and they say:

> Creates a new iterator that will apply the specified function to each element returned by the first, yielding the mapped element instead.

Does that mean that count() is going to be called on every iterator? Or how does that work in this case?